### PR TITLE
Generator for json definition of nodes

### PR DESCRIPTION
**Pull request by eparis**
node-gen --labels=rack=12,geo=Raleigh machine1.example.com machine2.example.com

Should create a json representation of a node which can be loaded into the apiserver

### CODE REVIEW

**Comment by eparis** [comment]
@bgrant0607 @mattf 

Thoughts?

**Comment by eparis** [comment]
It is a little annoying that the smallest 'generatable' json is:

`$ node-gen host.example.com`

```json
{
  "kind": "List",
  "apiVersion": "v1beta3",
  "metadata": {},
  "items": [
    {
      "kind": "Node",
      "apiVersion": "v1beta3",
      "metadata": {
        "name": "host.example.com",
        "creationTimestamp": null
      },
      "spec": {},
      "status": {
        "nodeInfo": {
          "machineID": "",
          "systemUUID": "",
          "bootID": "",
          "kernelVersion": "",
          "osImage": "",
          "containerRuntimeVersion": "",
          "kubeletVersion": "",
          "kubeProxyVersion": ""
        }
      }
    }
  ]
}
```

**Comment by eparis** [comment]
makes we wish for omitempty on creationTimestamp, spec, status....

**Comment by roberthbailey** [comment]
@eparis I wonder if this will be necessary after #6949 merges? It obsoletes the need for kube-register and may solve your use case as well.

File: contrib/node-gen/node-gen.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s\n", usage)
+		flag.PrintDefaults()
+	}
+}
+
+type image struct {
+	Host      string
+	Namespace string
+	Image     string
```
Comment (bgrant0607) [inline contrib/node-gen/node-gen.go]: Image? Where would the image come from? You're using this as the node name?

File: contrib/node-gen/node-gen.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			Labels: labelmap,
+		}
+		node := api.Node{
+			TypeMeta:   typeMeta,
+			ObjectMeta: metadata,
+		}
+		list.Items = append(list.Items, node)
+	}
+
+	jsonBytes, err := json.Marshal(list)
```
Comment (bgrant0607) [inline contrib/node-gen/node-gen.go]: If you're not going to call the Codec, you should be using pkg/api/v1beta3/types.go.
#4874
#3955

**Comment by bgrant0607** [comment]
Re. omitempty. Agree: https://github.com/GoogleCloudPlatform/kubernetes/issues/2726#issuecomment-67681313

**Comment by bgrant0607** [comment]
This could be useful, but we'd probably need more documentation about scenarios where it would work. 

It is also the case that self-registration may make this moot.

File: contrib/node-gen/node-gen.go (hunk 1 above)
Comment (eparis) [inline contrib/node-gen/node-gen.go]: left over junk from cp -a podex/ gen-node/ I forgot to delete.

**Comment by eparis** [comment]
@bgrant0607 I switched to using Codec, just to learn something new. Hopefully I did it correctly.  I'll add a README

File: contrib/node-gen/node-gen.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/latest"
+)
+
+const usage = "podex [--labels=LABELS] NODES..."
```
Comment (bgrant0607) [inline contrib/node-gen/node-gen.go]: podex? :-)

**Comment by eparis** [comment]
I signed it.

**Comment by eparis** [comment]
I signed it!

**Comment by eparis** [comment]
awww, the google bot got it in another pr  :-(

**Comment by eparis** [comment]
closing. auto registration mostly made me not care.

**Pull request closed without merging**